lib/shopping_cart.rb

```diff
@@ -0,0 +1,62 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details_hash = {
+      name: @name,
+      capacity: capacity
+    }
+  end
+
+  def total_number_of_products
+    quantity_total = 0
+    products.each do |product|
+      quantity_total += product.quantity
+    end
+    
+    quantity_total
+  end
+
+  def is_full?
+   total_number_of_products > capacity 
```

Comment:
i like that you're using your old method here to take care of this functionality